src/main/java/com/redhat/cloud/policies/app/auth/HeaderHelper.java

```diff
@@ -42,8 +42,8 @@ public static Optional<XRhIdentity> getRhIdFromString(String xRhIdHeader) {
         XRhIdentity rhIdentity;
         try {
             String json_string = new String(Base64.getDecoder().decode(xRhIdHeader));
-            rhIdentity = jsonb.fromJson(json_string, XRhIdentity.class);
-        } catch (JsonbException jbe) {
+            rhIdentity = objectMapper.readValue(json_string, XRhIdentity.class);
+        } catch (JsonProcessingException jpe) {
             return Optional.empty();
```

Comment:
It's worth to log this exception. This is happening when (for some reason) the identity header can't be parsed. It should succeed all the time unless there is a change that is affecting the parsing.

```java
import io.quarkus.logging.Log;

Log.error("Error while parsing the identity header", jpe);
```

I suppose it should be in a separate commit?

Reply:
I've stumbled on this whilst testing. It should be logged. Let's do that separately. Thank you.